homeassistant/components/voip/assist_satellite.py

```diff
@@ -332,6 +357,23 @@ async def async_start_conversation(
     # VoIP
     # -------------------------------------------------------------------------
 
+    def disconnect(self):
+        """Server disconnected."""
+        super().disconnect()
+        if self._check_hangup_task is not None:
```

Comment:
Should we also cancel the task when it's not done?

Reply:
Yes, added.